x-pack/plugin/ingest/src/main/java/org/elasticsearch/xpack/ingest/CommunityIdProcessor.java

```diff
@@ -0,0 +1,571 @@
+        String sourcePortField,
+        String destinationIpField,
+        String destinationPortField,
+        String ianaNumberField,
+        String transportField,
+        String icmpTypeField,
+        String icmpCodeField,
+        String targetField,
+        MessageDigest messageDigest,
+        byte[] seed,
+        boolean ignoreMissing
+    ) {
+        super(tag, description);
+        this.sourceIpField = sourceIpField;
+        this.sourcePortField = sourcePortField;
+        this.destinationIpField = destinationIpField;
+        this.destinationPortField = destinationPortField;
+        this.ianaNumberField = ianaNumberField;
+        this.transportField = transportField;
+        this.icmpTypeField = icmpTypeField;
+        this.icmpCodeField = icmpCodeField;
+        this.targetField = targetField;
+        this.messageDigest = messageDigest;
+        this.seed = seed;
+        this.ignoreMissing = ignoreMissing;
+    }
+
+    public String getSourceIpField() {
+        return sourceIpField;
+    }
+
+    public String getSourcePortField() {
+        return sourcePortField;
+    }
+
+    public String getDestinationIpField() {
+        return destinationIpField;
+    }
+
+    public String getDestinationPortField() {
+        return destinationPortField;
+    }
+
+    public String getIanaNumberField() {
+        return ianaNumberField;
+    }
+
+    public String getTransportField() {
+        return transportField;
+    }
+
+    public String getIcmpTypeField() {
+        return icmpTypeField;
+    }
+
+    public String getIcmpCodeField() {
+        return icmpCodeField;
+    }
+
+    public String getTargetField() {
+        return targetField;
+    }
+
+    public MessageDigest getMessageDigest() {
+        return messageDigest;
+    }
+
+    public byte[] getSeed() {
+        return seed;
+    }
+
+    public boolean getIgnoreMissing() {
+        return ignoreMissing;
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
+        Flow flow = buildFlow(ingestDocument);
+        if (flow == null) {
+            if (ignoreMissing) {
+                return ingestDocument;
+            } else {
+                throw new IllegalArgumentException("unable to construct flow from document");
+            }
+        }
+
+        ingestDocument.setFieldValue(targetField, flow.toCommunityId(messageDigest, seed));
+        return ingestDocument;
+    }
+
+    private Flow buildFlow(IngestDocument d) {
+        String sourceIpAddrString = d.getFieldValue(sourceIpField, String.class, ignoreMissing);
+        if (sourceIpAddrString == null) {
+            return null;
+        }
+
+        String destIpAddrString = d.getFieldValue(destinationIpField, String.class, ignoreMissing);
+        if (destIpAddrString == null) {
+            return null;
+        }
+
+        Flow flow = new Flow();
+        try {
+            flow.source = InetAddress.getByName(sourceIpAddrString);
```

Comment:
I'm worried that this could potentially perform network calls to lookup IP addresses from names.

Luckily we have a utility for this in https://github.com/elastic/elasticsearch/blob/feab123ba400b150f3dcd04dd27cf57474b70d5a/server/src/main/java/org/elasticsearch/common/network/InetAddresses.java#L340 that's safe to use.